### PR TITLE
Add: Console command to list search directories for various things

### PR DESCRIPTION
**Pull request by nielsmh**
## Motivation / Problem

It can be difficult to figure out exactly where to put various files for the game to find them, especially if you're running a headless (dedicated) server and need to upload files by SFTP or similar.

## Description

This adds a console command that lists all the search directories for all the interesting game data files, as well as the default locations for save, autosave and screenshot files.

image

## Limitations

Potentially leaks some system information when used over remote console, should this have a protection against that?

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/console_cmds.cpp [hunk 1]
```diff
@@ -1975,6 +1975,64 @@ DEF_CONSOLE_CMD(ConNewGRFReload)
...
+			if (!first) cats = cats + ", " + sdn.name;
+			first = false;
+		}
+		IConsolePrint(CC_HELP, "Valid categories: {}", cats);
+		return true;
+	}
+
+	std::set<std::string> seen_dirs;
+	for (const SubdirNameMap &sdn : subdir_name_map) {
+		if (strcasecmp(argv[1], sdn.name) == 0) {
```
Comment (TrueBrain) [inline src/console_cmds.cpp]: `!=` and continue? Lowers the nesting a bit.

File: src/console_cmds.cpp [hunk 2]
```diff
@@ -1975,6 +1975,64 @@ DEF_CONSOLE_CMD(ConNewGRFReload)
...
+			bool found = false;
+			for (Searchpath sp : _valid_searchpaths) {
```
Comment (TrueBrain) [inline src/console_cmds.cpp]: Not const or `&`?

File: src/console_cmds.cpp [hunk 3]
```diff
@@ -1975,6 +1975,64 @@ DEF_CONSOLE_CMD(ConNewGRFReload)
...
+		{ GAME_LIBRARY_DIR, "gslib",      false },
+		{ SAVE_DIR,         "save",       true  },
+		{ AUTOSAVE_DIR,     "autosave",   true  },
+		{ SCENARIO_DIR,     "scenario",   false },
+		{ HEIGHTMAP_DIR,    "heightmap",  false },
+		{ SCREENSHOT_DIR,   "screenshot", true  },
+	};
+
+	if (argc != 2) {
+		IConsolePrint(CC_HELP, "List all search paths or default paths for various categories.");
```
Comment (TrueBrain) [inline src/console_cmds.cpp]: Left over `paths` ;)

File: src/console_cmds.cpp [hunk 4]
```diff
@@ -1975,6 +1975,64 @@ DEF_CONSOLE_CMD(ConNewGRFReload)
 	return true;
 }
 
+DEF_CONSOLE_CMD(ConListDirs)
+{
+	struct SubdirNameMap {
+		Subdirectory subdir;
+		const char *name;
+		bool default_only;
```
Comment (TrueBrain) [inline src/console_cmds.cpp]: Comment to explain what this does? As I am not 100% sure ;)

**Review by TrueBrain — COMMENTED** [review]
Mostly nitpicks :)

File: src/console_cmds.cpp (hunk 2 above)
Comment (nielsmh) [reply src/console_cmds.cpp]: It's literally just an enum constant, making it a reference will gain nothing, but potentially make the compiler generate worse code. (At least in my head.)

**Review by TrueBrain — APPROVED** [review]

**Pull request merged**